### DISCUSSION TITLE
How to configure Docker-ized pushgateway ?

### DISCUSSION THREAD

**Discussion by Gaibhne** [prometheus/pushgateway#495] [Q&A]
Various things in the documentation suggest that you can configure the pushgateway (such as the persistence interval), but I have been unable to figure out where, if anywhere, that is documented.

Any help ?

**Comment by beorn7** [comment]
I would say you this should be possible by the usual Docker means (of which I'm not an expert, so I better leave a conclusive answer to others).

**Comment by Gaibhne** [comment]
I'm quite comfortable with Docker, but other than two mentioned arguments to the pushgateway, I found no other documentation as to possible arguments, their syntax, valid ranges of settings and so on. Neither argument related to the details I was interested in that were mentioned in the documentation such as external URL.

There is no generic 'Docker way' of passing arguments, sometimes they can be supplied via `command`, other images have entry point scripts that parse env variables, yet again others read in config files, so a generic Docker answer would not be helpful.

**Reply by beorn7** [reply]
I just tried it out and simply ran
```
docker run --rm -it -p 9091:9091 prom/pushgateway:v1.4.1 --persistence.file=persist.db --persistence.interval=5m
```

As said, I'm not a Docker expert, but this seems to me to be a pretty vanilla way of running a container. The command line arguments make it through to the executed command just fine.

**Comment by Gaibhne** [comment]
Thanks, that answers the first part of the question.

To the second part, it adds a third argument (`--persistence.interval`) to the two mentioned in the README that I can use, but aren't these documented anywhere ? Is the only way to find out about them to read the source ?

**Comment by beorn7** [comment]
You can simply run `pushgateway --help`. Or even that could be run from within the container: `docker run --rm -it prom/pushgateway:v1.4.1 --help`
